core/FrontController.php

```diff
@@ -392,18 +392,21 @@ public function init()
 
         $loggedIn = false;
 
-        // try authenticating w/ session first...
-        $sessionAuth = $this->makeSessionAuthenticator();
-        if ($sessionAuth) {
-            $loggedIn = Access::getInstance()->reloadAccess($sessionAuth);
-        }
+        // user may already be logged in when init() is called, eg, in a CLI command
+        if (!Access::getInstance()->isUserLoggedIn()) {
+            // try authenticating w/ session first...
+            $sessionAuth = $this->makeSessionAuthenticator();
```

Comment:
@diosmosis I'm not sure re possible side effects or so. 

I wonder... if technically someone could do something like this 

```
$_GET['token_auth'] = ...
FrontController::getInstance()->dispatch()
```

if that could be any issue. Or if the user may be temporarily logged in using `doAsSuperUser` but then the `Access::$login` is not reset or so. 

It might be a great fix I'm just unable to tell re side effects etc it's so tricky! In generally it does sound good though cause there should be a need to authenticate again. Unless maybe for some random use case, not sure. I reckon we could do this only on CLI and maybe only skip the session auth? Or always skip session auth on CLI since there is no session? Might be more "secure" in terms of possible side effects?

Reply:
I guess it makes sense to just skip sessionauth if in cli mode. Can't imagine how it would work then...

Reply:
@tsteur sound good to change it to:

```
if (!Access::getInstance()->isPhpCliMode()) {
    ...
```

?

Reply:
👍  Although I reckon you mean `Common::isPhpClimode`?

Reply:
Yes :) don't know how that got past me :)